indigo_app/static/javascript/indigo/grammars/slaw.js

```diff
@@ -0,0 +1,175 @@
+(function(exports) {
+  "use strict";
+
+  class SlawGrammarModel extends Indigo.grammars.GrammarModel {
+    constructor (...args) {
+      super(...args);
+      this.language_id = 'slaw';
+      this.language_def = {
+        defaultToken: '',
+        hier: /Part|Chapter|Subpart/,
+        markers: /PREFACE|PREAMBLE|BODY/,
+        headings: /LONGTITLE|CROSSHEADING|HEADING|SUBHEADING/,
+        tokenizer: {
+          root: [
+            // grouping markers
+            [/^@markers\s*$/, 'keyword.marker'],
+
+            // single line headings
+            [/^(\s*)(@headings)(\s.*)$/, ['white', 'keyword.heading', 'string']],
+
+            // hierarchical
+            // Part num - heading
+            [/^(\s*)(@hier)(\s+.+)(-)(\s+.*$)/i, ['white', 'keyword.hier', 'constant.numeric', 'delimiter', 'string']],
+            // Part num
+            [/^(\s*)(@hier)(\s+.+$)/i, ['white', 'keyword.hier', 'constant.numeric']],
+
+            // 2. Section heading
+            [/^(\s*)([0-9][0-9a-z]*\.)(\s.*$)/, ['white', 'constant.numeric', 'string']],
```

Comment:
should the regex end in `/i` for e.g. section 2A, or alternatively `a-z` -> `A-Z`?

Reply:
Added